### PR TITLE
[enable-load-rosbag] Add a feature that generates streams from RosBag 

### PR DESCRIPTION
**Pull request by wooksong**
This patch adds a feature that generates Tensor(s) streams from RosBag instead of the actual topic subscription.

Signed-off-by: Wook Song <wook16.song@samsung.com>

### CODE REVIEW

File: ros2/node/nns_rclcpp_subscriber.cc [hunk 1]
```diff
@@ -247,7 +298,10 @@ void NnsRclCppSubscriber::loop ()
   );
 
   while (true) {
-    rclcpp::spin_some (this->node_);
+    if (!this->is_dummy)
+      rclcpp::spin_some (this->node_);
+    else
+      this->fetchDataFromBag ();
```
Comment (myungjoo) [inline ros2/node/nns_rclcpp_subscriber.cc:304]: Who's using the lock ```lk``` of line 307?

File: ros/node/nns_roscpp_subscriber.cc [hunk 2]
```diff
@@ -262,17 +276,47 @@ NnsRosCppSubscriber::loop ()
...
+    }
+  } else {
+    while (true) {
+      ros::spinOnce ();
+      ros_rate.sleep ();
+      {
       std::unique_lock<std::mutex> lk (this->g_m);
       if (!this->is_looper_ready)
         break;
+      }
```
Comment (myungjoo) [inline ros/node/nns_roscpp_subscriber.cc:314]: Strange indentation

File: ros2/node/nns_rclcpp_subscriber.cc (hunk 1 above)
Comment (wooksong) [reply ros2/node/nns_rclcpp_subscriber.cc:304]: @myungjoo That is a guard for the finalize method (at line 187) that would run on another thread.

File: ros/node/nns_roscpp_subscriber.cc (hunk 2 above)
Comment (wooksong) [reply ros/node/nns_roscpp_subscriber.cc:314]: @myungjoo Right. I will fix it.

File: ros2/node/nns_rclcpp_subscriber.cc (hunk 1 above)
Comment (myungjoo) [reply ros2/node/nns_rclcpp_subscriber.cc:304]: That is a constructor. Don't you need to call ```std::lock(lk);``` in the scope of ```lk```?

Comment (wooksong) [reply ros2/node/nns_rclcpp_subscriber.cc:304]: @myungjoo IMHO, ```unique_lock``` does not require invoking any ```std::lock``` method families. It accquires the given mutex, i.e., ```this->g_m``` on construction and releases that mutex on destruction.

Comment (again4you) [reply ros2/node/nns_rclcpp_subscriber.cc:304]: IIRC, `std::unique_lock()` basically works like `std::lock_guard()`.
However, if `std::defer_lock` option is used when calling `std::unique_lock()`, `std::lock()` is explicitly necessary to lock the critical section.

Comment (myungjoo) [reply ros2/node/nns_rclcpp_subscriber.cc:304]: Ah.. this locks when declared and unlocks when it goes out of the block. Great!

**Review by myungjoo — APPROVED** [review]
LGTM

**Review by again4you — APPROVED** [review]
LGTM!

**Pull request merged**